File: CS2242_SMART_WHEELCHAIR/src/system/WheelChair.java
package system;

public class WheelChair {
	private int id;
	private Location defaultLocation;
	private Location currentLocation;
	private float distanceTravelled;
//	private boolean flipped;
	
	public WheelChair(int id, Location defaultLocation) {
		this.id = id;
		this.defaultLocation = defaultLocation;
	}

	protected Location getDefaultLocation() {
		return defaultLocation;
	}

	protected void setDefaultLocation(Location defaultLocation) {
		this.defaultLocation = defaultLocation;
	}

	protected Location getCurrentLocation() {
		return currentLocation;
	}

	protected void setCurrentLocation(Location currentLocation) {
		this.currentLocation = currentLocation;
	}

	protected int getId() {
		return id;
	}

	protected float getDistanceTravelled() {
		return distanceTravelled;
	}
	
}

File: Thilina/C++ Classes/wheel_chair.cpp
#include"wheel_chair.h"

WheelChair::WheelChair(const std::string& id, const std::string& default_location) : id(id), default_location(default_location){

} 

//setters
void WheelChair::change_patient(const Patient& new_patient){}
void WheelChair::change_default_location(const std::string& new_default_location){}
void WheelChair::chage_current_location(const std::string& current_location){}

//getters
const std::string& WheelChair::get_id() const{}
const Patient& WheelChair::get_patient() const{}
const std::string& WheelChair::get_default_location() const{}
const std::string& WheelChair::get_current_location() const{}